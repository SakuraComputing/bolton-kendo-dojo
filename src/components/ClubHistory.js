import React from 'react'
import { connect } from 'react-redux';
import Spinner from '../common/Spinner';

export class ClubHistory extends React.Component {

  render() {

    let clubHistory;

    const { club, loading } = this.props.club;

    if(club !== null && !loading) {
      clubHistory = (
        <div>{club[0].clubHistory}</div>
      )
    } else {
      clubHistory = <Spinner />
    }

    return (
      <div>
        <h1>Club History</h1>
        {clubHistory}
      </div>
    )
  }
}
const mapStateToProps = (state) => ({
  club: state.club
})
export default connect(mapStateToProps)(ClubHistory);
