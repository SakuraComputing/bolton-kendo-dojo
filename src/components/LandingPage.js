import React from 'react'

const LandingPage = () => {
  return (
    <div>
        <h1>Bolton Kendo Dojo</h1>
    </div>
  )
}
export default LandingPage;
