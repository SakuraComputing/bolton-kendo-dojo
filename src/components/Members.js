import React from 'react';
import { Link } from 'react-router-dom';

const  Members = () => {
  return (
    <div className="members">
      <h1 className="members-header">Welcome to the Members Area</h1>
      <div className="members-area-container">
        <div className="members-area-link">
          <Link to="/memberphoto">Photographs</Link>
          <p>Select the photgraphs to go in to the photgraph section</p>
        </div>
        <div className="members-area-link">
          <Link to="/memberphoto">Kendo Mind Map</Link>
          <p>Enter the Kendo Mind Map section for all deeper kendo study</p>
        </div>
        <div className="members-area-link">
          <Link to="/memberphoto">Videos</Link>
          <p>Go into the video section and view private members videos</p>
        </div>
        <div className="members-area-link">
          <Link to="/members">Bulletin Board</Link>
          <p>Add a post to the bulletin board. Responds to other posts</p>
        </div>
      </div>
    </div>
  )
}
export default Members;
