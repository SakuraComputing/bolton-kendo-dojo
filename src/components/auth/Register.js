import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import TextFieldGroup from '../../common/TextFieldGroup';
import { registerUser } from '../../actions/authActions';

export class Register extends Component {

    state = {
        name: '',
        email: '',
        password: '',
        password2: '',
        errors: {}
    };

    onChange = (e) => {
        this.setState({[e.target.name]: e.target.value})
    };

    onSubmit = (e) => {
        e.preventDefault();
        const newUser = {
            name: this.state.name,
            email: this.state.email,
            password: this.state.password,
            password2: this.state.password2
        };
        this.props.registerUser(newUser, this.props.history);
    };

    render() {
        const { errors } = this.state;
        return (
            <div className="register">
                <div className="container">
                    <h1 className="form-title">Sign Up</h1>
                    <p className="form-text">Create your Members Account</p>
                    <form noValidate onSubmit={this.onSubmit}>
                        <TextFieldGroup
                            placeholder="Name"
                            name="name"
                            value={this.state.name}
                            type="name"
                            onChange={this.onChange}
                            error={errors.name}
                        />
                        <TextFieldGroup
                            placeholder="Email Address"
                            name="email"
                            value={this.state.email}
                            type="email"
                            onChange={this.onChange}
                            error={errors.email}
                        />
                        <TextFieldGroup
                            placeholder="Password"
                            name="password"
                            value={this.state.password}
                            type="password"
                            onChange={this.onChange}
                            error={errors.password}
                        />
                        <TextFieldGroup
                            placeholder="Confirm Password"
                            name="password2"
                            value={this.state.password2}
                            type="password"
                            onChange={this.onChange}
                            error={errors.password2}
                        />

                        <input type="submit" className="button button-info"/>
                    </form>
                </div>
            </div>
        );
    }
}

Register.propTypes = {
    registerUser: PropTypes.func.isRequired,
    auth: PropTypes.object.isRequired,
    errors: PropTypes.object.isRequired
};

const mapStateToProps = (state) => ({
    auth: state.auth,
    errors: state.errors
});

export default connect(mapStateToProps, { registerUser })(withRouter(Register));
