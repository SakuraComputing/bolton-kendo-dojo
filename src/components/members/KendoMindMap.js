import React, { Component } from 'react';

class KendoMindMap extends Component {

  render() {
    return (
      <div className="content-container mindmap-layout">
        <img src="/img/background/KendoMindMap.png" alt=""/>
      </div>
    )
  }
}
export default KendoMindMap;
