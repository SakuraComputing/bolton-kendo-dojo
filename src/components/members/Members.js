import React from 'react';
import { Link } from 'react-router-dom';

const  Members = () => {
  return (
    <div className="members">
      <h1 className="members-header">Welcome to the Members Area</h1>
      <div className="members-area-container">
        <div className="members-area-link">
          <Link to="/memberphoto">Photos</Link>
          <p>Select the photographs and upload these with text for other members to see.</p>
        </div>
        <div className="members-area-link">
          <Link to="/mindmap">Kendo Mind Map</Link>
          <p>Enter the Kendo Mind Map section for all deeper kendo study</p>
        </div>
        <div className="members-area-link">
          <Link to="/memberphoto">Videos</Link>
          <p>Go into the video section and view private members videos</p>
        </div>
        <div className="members-area-link">
          <Link to="/members">Bulletin Board</Link>
          <p>Add a post to the bulletin board. Respond to other posts</p>
        </div>
      </div>
    </div>
  )
}
export default Members;
