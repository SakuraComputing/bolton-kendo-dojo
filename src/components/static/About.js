import React from 'react';
import { connect } from 'react-redux';
import Map, { Marker } from 'react-map-gl';
import Pin from '../../common/Pin';
import Spinner from '../../common/Spinner';

const key = require('../../config/keys').apiMapKey;
const test = require('../../config/keys').mongoURI;

const API_KEY = key;

console.log('Map key', key, test);

const LATITUDE = 53.562614;
const LONGITUDE = -2.384833;

export class About extends React.Component {

  constructor(props) {
    super(props);
      this.state = {
        viewport: {
            latitude: LATITUDE,
            longitude: LONGITUDE,
            zoom: 14,
            bearing: 0,
            pitch: 0,
            width: '100%',
            height: 400,
        }
    };
  }

  render() {

    let clubContent, practises, contacts, address;

    const { viewport } = this.state;

    const { club, loading } = this.props.club;

    if(club !== null && !loading) {
        practises = club[0].practices.map(practise => (
            <div key={key}>
                <div>{practise.venue}</div>
                <div>{practise.time}</div>
            </div>
        ));
        contacts = club[0].contacts.map(contact => (
            <div key={key}>
                <div>{contact.name}</div>
                <div>{contact.telephoneNumber}</div>
                <div>{contact.email}</div>
            </div>
        ));
        address = club[0].addresses.map(address => (
            <div key={key}>
                <div>{address.address1}</div>
                <div>{address.address2}</div>
                <div>{address.address3}</div>
                <div>{address.address4}</div>
                <div>{address.postcode.slice(0,3) + " " + address.postcode.slice(3,6)}</div>
            </div>
        ));
        clubContent = (
            <div>
                <div className="about-map-section" >
                    <div className="about-header">
                        <div className="about-section">
                            <h2>Where we practice</h2>
                            {address}
                        </div>
                        <div className="about-section">
                            <h2>Practice Times</h2>
                            {practises}
                        </div>
                        <div className="about-section" >
                            <h2>Contact</h2>
                            {contacts}
                        </div>
                    </div>
                    <div className="about-map">
                        <Map
                            {...viewport}
                            mapStyle="mapbox://styles/mapbox/streets-v10"
                            mapboxApiAccessToken={API_KEY}>
                            <Marker 
                                latitude={LATITUDE} 
                                longitude={LONGITUDE}
                            >
                                <Pin />       
                            </Marker>
                        </Map>
                        
                    </div>
                </div>
                <div>
                </div>
            </div>
        )
    } else {
        clubContent = <Spinner />
    }
    
    return (
        <div className="about-container">
            {clubContent}
        </div>
    )
  }
}

const mapStateToProps = (state) => ({
    club: state.club
});

export default connect(mapStateToProps)(About);
