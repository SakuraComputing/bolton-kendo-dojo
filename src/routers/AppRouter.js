import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';



import LandingPage from '../components/LandingPage';
import Login from '../components/auth/login';
import Menu from '../components/Menu';
import Footer from '../components/Footer';
import Register from '../components/auth/Register';
import TheoryPage from '../components/static/TheoryPage';
import KendoLinks from '../components/static/KendoLinks';
import NotFound from '../components/static/NotFoundPage';
import Members from '../components/Members';
import About from '../components/static/About';

import '../styles/styles.scss';
import PrivateRoute from '../common/PrivateRoute';

class AppRouter extends Component {

    render() {
        return (
            <BrowserRouter>
                <div className="App">
                    <Menu />
                    <Switch>
                        <Route path="/" component={LandingPage} exact={true}/>
                        <Route exact path="/login" component={Login}></Route>
                        <Route exact path="/register" component={Register}></Route>
                        <Route exact path="/theory" component={TheoryPage}></Route>
                        <Route exact path="/links" component={KendoLinks}></Route>
                        <Route exact path="/about" component={About}></Route>
                        <Route path="/not-found" component={NotFound}/>
                    </Switch>
                    <Switch>
                        <PrivateRoute
                            exact path="/members"
                            component={Members}
                        />
                    </Switch>
                    <Footer />
                </div>
            </BrowserRouter>
        )
    }
};

export default AppRouter;