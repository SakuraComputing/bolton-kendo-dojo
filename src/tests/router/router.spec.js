import React from 'react';
import AppRouter from '../../routers/AppRouter';
import { shallow } from 'enzyme';
import LandingPage from '../../components/LandingPage';
import Login from '../../components/auth/Login';
import Register from '../../components/auth/Register';
import Theory from '../../components/static/TheoryPage';
import KendoLinks from '../../components/static/KendoLinks';
import About from '../../components/static/About';
import ClubHistory from '../../components/ClubHistory';
import Members from '../../components/members/Members';
import KendoMindMap from '../../components/members/KendoMindMap';
import NotFound from '../../components/static/NotFoundPage';

describe('App Router', () => {
    it('AppRouter is a function', () => {
        expect(typeof AppRouter).toEqual('function');
    })

    let wrapper; 

    beforeEach(() => {
        wrapper = shallow(<AppRouter />);
    })
    it('renders as expected', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should route to the landing page', () => {
        expect(wrapper.find('Route[exact=true][path="/"]').first().prop('component')).toBe(LandingPage);
    });
    it('should route to the login page', () => {
        expect(JSON.stringify(wrapper.find('Route[path="/login"]').first().prop('component'))).toEqual(JSON.stringify(Login));
    });
    it('should route to the register page', () => {
        expect(wrapper.find('Route[path="/register"]').first().prop('component')).toEqual(Register);
    });
    it('should route to the Theory page', () => {
        expect(wrapper.find('Route[path="/theory"]').first().prop('component')).toEqual(Theory);
    });
    it('should route to the Links page', () => {
        expect(wrapper.find('Route[path="/links"]').first().prop('component')).toEqual(KendoLinks);
    });
    it('should route to the not found page', () => {
        expect(wrapper.find('Route[path="/not-found"]').first().prop('component')).toEqual(NotFound);
    });
    it('should route to the about page', () => {
        expect(wrapper.find('Route[path="/about"]').first().prop('component')).toEqual(About);
    });
    it('should route to the club history page', () => {
        expect(wrapper.find('Route[path="/history"]').first().prop('component')).toEqual(ClubHistory);
    });
    it.skip('should route to the kendo members page', () => {
        expect(wrapper.find('PrivateRoute[path="/members"]').first().prop('component')).toEqual(Members);
    });
    it.skip('should route to the kendo mind map page', () => {
        expect(wrapper.find('Route[path="/mindmap"]').first().prop('component')).toEqual(KendoMindMap);
    });

})