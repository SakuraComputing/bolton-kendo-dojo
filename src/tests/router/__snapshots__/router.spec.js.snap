// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Router renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppRouter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="app-container"
      >
        <Connect(Menu) />
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/login"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/register"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/theory"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/links"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/about"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/history"
          />
          <Route
            component={[Function]}
            path="/not-found"
          />
        </Switch>
        <Switch>
          <Connect(PrivateRoute)
            component={[Function]}
            exact={true}
            path="/members"
          />
        </Switch>
        <Switch>
          <Connect(PrivateRoute)
            component={[Function]}
            exact={true}
            path="/mindmap"
          />
        </Switch>
        <Switch>
          <Connect(PrivateRoute)
            component={[Function]}
            exact={true}
            path="/memberphoto"
          />
        </Switch>
        <Footer />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Menu) />,
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/register"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/theory"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/links"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/about"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/history"
            />
            <Route
              component={[Function]}
              path="/not-found"
            />
          </Switch>,
          <Switch>
            <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/members"
            />
          </Switch>,
          <Switch>
            <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/mindmap"
            />
          </Switch>,
          <Switch>
            <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/memberphoto"
            />
          </Switch>,
          <Footer />,
        ],
        "className": "app-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/register"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/theory"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/links"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/about"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/history"
              />,
              <Route
                component={[Function]}
                path="/not-found"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/register",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/theory",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/links",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/about",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/history",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/not-found",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/members"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/members",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/mindmap"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/mindmap",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/memberphoto"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/memberphoto",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="app-container"
        >
          <Connect(Menu) />
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/register"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/theory"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/links"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/about"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/history"
            />
            <Route
              component={[Function]}
              path="/not-found"
            />
          </Switch>
          <Switch>
            <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/members"
            />
          </Switch>
          <Switch>
            <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/mindmap"
            />
          </Switch>
          <Switch>
            <Connect(PrivateRoute)
              component={[Function]}
              exact={true}
              path="/memberphoto"
            />
          </Switch>
          <Footer />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(Menu) />,
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/register"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/theory"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/links"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/about"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/history"
              />
              <Route
                component={[Function]}
                path="/not-found"
              />
            </Switch>,
            <Switch>
              <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/members"
              />
            </Switch>,
            <Switch>
              <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/mindmap"
              />
            </Switch>,
            <Switch>
              <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/memberphoto"
              />
            </Switch>,
            <Footer />,
          ],
          "className": "app-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/login"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/register"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/theory"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/links"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/about"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/history"
                />,
                <Route
                  component={[Function]}
                  path="/not-found"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/login",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/register",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/theory",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/links",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/about",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/history",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/not-found",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/members"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/members",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/mindmap"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/mindmap",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(PrivateRoute)
                component={[Function]}
                exact={true}
                path="/memberphoto"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/memberphoto",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
